api-changes.md

```diff
@@ -3,6 +3,120 @@
 Informal notes ahead of the release. This document will be merged
 into HISTORY when the release is cut.
 
+## AST changes
+
 - Remove non-device selectors.
 
+- Table, Stream, Actions are no longer used. Instead, everything is
+  an Expression, which could be a table Expression (Index, Sort, Aggregation),
+  a table-or-stream Expression (Filter, Projection), or any of the
+  three types (FunctionCall, Invocation)
+
+- The Expression class has a toLegacy() method to convert to the old AST
+  objects and aid in the transition.
+
+- NotifyActions are no longer used.
+
+- To parse the new syntax, use `Grammar.parse()` and pass `SyntaxType.Normal`
+  (user input) or `SyntaxType.Tokenized` (neural network output)
+  
+- To serialize to the new syntax, use `node.toSource()` to generate a token
+  stream (for neural network) and `node.prettyprint2()` to generate actual syntax
+
+## Syntax changes compared to old surface syntax
+
+### Major
+
+- `now` and `notify` keyword no longer exist (they are still accepted to aid
+  transition but are no longer required)
+
+- aggregate syntax is now `op(field of table)`; example:
+  ```
+  min(file_size of @com.dropbox.list_folder())
+  ```
+  
+  `count` is just that:
+  ```
+  count(@com.dropbox.list_folder())
+  ```
+
+- sort syntax is now `sort(field direction of table)`; example:
+  ```
+  sort(file_size asc of @com.dropbox.list_folder())
+  ```
+  
+- computation is now part of projection, with the syntax `[expr1, expr2, ...] of table`; example:
+  ```
+  [geo, cuisines, distance(geo, $location.current_location)] of @com.yelp.restaurant()
+  ```
+  
+- computation can be aliased; example:
+  ```
+  [distance(geo, $location.current_location) as dist] of @com.yelp.restaurant()
+  ```
+
+- `join` is now expressed as `=>` (true table joins should be expressed as
+  subqueries instead)
+  
+- `monitor` of a specific field is `monitor(fields of table)`; example:
+  ```
+  monitor(text, author of @com.twitter.home_timeline())
```

Comment:
It's a little inconsistent here between monitor and projection. could we remove the `[]` in projection? otherwise, we should probably add `[]` here.